msbuild/Xamarin.Shared/Xamarin.Shared.targets

```diff
@@ -1183,6 +1183,203 @@ Copyright (C) 2018 Microsoft. All rights reserved.
 		/>
 	</Target>
 
+	<Target Name="_GetNativeExecutableName" DependsOnTargets="_DetectAppManifest;_GenerateBundleName;_CompileAppManifest">
+		<GetNativeExecutableName
+			SessionId="$(BuildSessionId)"
+			Condition="'$(IsMacEnabled)' == 'true'"
+			AppManifest="$(_AppBundlePath)$(_AppBundleManifestRelativePath)Info.plist"
+			>
+			<Output TaskParameter="ExecutableName" PropertyName="_ExecutableName" />
+		</GetNativeExecutableName>
+
+		<PropertyGroup>
+			<_NativeExecutable Condition="'$(_PlatformName)' != 'macOS'">$(_AppBundlePath)$(_ExecutableName)</_NativeExecutable>
+			<_NativeExecutable Condition="'$(_PlatformName)' == 'macOS' Or '$(_IsMacCatalyst)' == 'true'">$(_AppBundlePath)Contents\MacOS\$(_ExecutableName)</_NativeExecutable>
+		</PropertyGroup>
+	</Target>
+
+	<Target Name="_ReadAppExtensionDebugSymbolProperties">
+		<ReadItemsFromFile File="%(_ResolvedAppExtensionReferences.Identity)\..\dsym.items" Condition="Exists('%(_ResolvedAppExtensionReferences.Identity)\..\dsym.items')">
+			<Output TaskParameter="Items" ItemName="_AppExtensionDebugSymbolProperties" />
+		</ReadItemsFromFile>
+	</Target>
+
+	<Target Name="_GenerateAppExtensionDebugSymbols" Condition="'$(ComputedPlatform)' == 'iPhone' And '$(IsAppExtension)' == 'false'"
```

Comment:
The condition here will never include Xamarin.Mac - is that on purpose?

Reply:
Yes, app extensions differs between the platforms and this PR (originally for [xc]frameworks support) does not cover it.
I'll file an issue so it's not forgotten.

Reply:
https://github.com/xamarin/xamarin-macios/issues/10412